api/s2n.h

```diff
@@ -974,6 +974,30 @@ S2N_API extern int s2n_config_set_verify_host_callback(struct s2n_config *config
  */
 S2N_API extern int s2n_config_set_check_stapled_ocsp_response(struct s2n_config *config, uint8_t check_ocsp);
 
+/**
+ * Specifies whether to validate timestamp fields on received certificates.
+ *
+ * By default, s2n-tls checks the notBefore and notAfter fields on the certificates it receives
+ * during the handshake. If the current date is not within the range of these fields for any
+ * certificate in the chain of trust, `s2n_negotiate()` will error. This validation is in
+ * accordance with RFC 5280, section 6.1.3 a.2:
+ * https://datatracker.ietf.org/doc/html/rfc5280#section-6.1.3.
+ *
+ * This API will disable this timestamp validation, permitting negotiation with peers that send
+ * expired certificates, or certificates that are not yet considered valid.
+ *
+ * @warning Applications calling this API should seriously consider the security implications of
+ * disabling this validation. The validity period of a certificate corresponds to the range of time
+ * in which the CA is guaranteed to maintain information regarding the certificate's revocation
+ * status. As such, it may not be possible to obtain accurate revocation information for
+ * certificates with invalid timestamps.
```

Comment:
You're assuming that applications know they should be using CRLs when doing this :) That's probably worth a warning too-- this is even more unsafe without an alternative method (like CRLs) of limiting the lifetime of certificates. Usually if a certificate is compromised, the impact is timeboxed by its validity period.

Reply:
I added a sentence about applications making sure there's a way to limit cert lifetime. However, I'm hesitant to mention CRLs here specifically, since in the normal case where you get your CRL from the CA, the CA doesn't have to keep the expired certificates around anymore. So, CRLs are only useful if applications maintain them themselves, and explicitly don't remove expired certificates even though they're permitted to do so by the RFC.